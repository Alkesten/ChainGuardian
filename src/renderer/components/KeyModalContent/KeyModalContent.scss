.key-input-container {
    display: flex;
    flex-direction: column;
    width: 100%;
}

.mt-32 {
    margin-top: 32px;
}

.submit-button-container {
    margin-top: 40px;
    text-align: center;
}