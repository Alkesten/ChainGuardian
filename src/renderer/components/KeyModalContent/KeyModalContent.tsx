import React, {useState, ReactElement, useEffect} from "react";
import {InputForm} from "../Input/InputForm";
import {ButtonPrimary} from "../Button/ButtonStandard";
import {Eth2HDWallet} from "../../services/Eth2HDWallet";

interface IKeyModalProps {
    title: string,
    description?: string,
    onSubmit: (input: string) => void
}

export default function KeyModalContent(props: IKeyModalProps): ReactElement {
    const [input, setinput] = useState("");
    const [valid, setvalid] = useState(false);
    const [errorMessage, setErrorMessage] = useState("");

    const handleChange = (e: React.FormEvent<HTMLInputElement>): void => {
        const input = e.currentTarget.value
        setinput(input);

        // 
        if(input === ""){
            setvalid(false)
            setErrorMessage("")
            return
        }

        const {isValid, message} = Eth2HDWallet.checkValidity(input);
        setvalid(isValid);

        if(!isValid){
            setErrorMessage(message)
        }
    };

    const handleSubmit = (): void => {
        if(valid){
            props.onSubmit(input);
        }
    };

    useEffect(() => {}, [valid]);

    return (
        <>
            <h1>{props.title}</h1>
            {props.description &&
                <p>
                    {props.description}
                </p>
            }
            <div className={`key-input-container ${props.description ? "" : "mt-32"}`}>
                <InputForm
                    inputId="inputKey"
                    focused
                    onChange={handleChange}
                    inputValue={input}
                    valid={errorMessage === "" && valid === false ? undefined : valid}
                    errorMessage={errorMessage}
                    placeholder="Enter your unique mnemonic signing key" />
                <span className="submit-button-container">
                    <ButtonPrimary buttonId="submit" onClick={handleSubmit}>Submit</ButtonPrimary>
                </span>
            </div>
        </>
    );
}
