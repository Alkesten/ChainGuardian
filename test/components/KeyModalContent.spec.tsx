import * as React from "react";
import * as renderer from "react-test-renderer";
import KeyModalContent from "../../src/renderer/components/KeyModalContent/KeyModalContent";

describe("KeyModalContent", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<KeyModalContent title="Test" onSubmit={(): void => { }}></KeyModalContent>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders description", () => {
        const tree = renderer
            .create(
                <KeyModalContent
                    title="Test"
                    onSubmit={(): void => { }}
                    description="Test description"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});