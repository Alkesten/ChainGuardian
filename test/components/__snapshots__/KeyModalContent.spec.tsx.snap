// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyModalContent renders correctly 1`] = `
Array [
  <h1>
    Test
  </h1>,
  <div
    className="key-input-container mt-32"
  >
    <form>
      <div
        className="label"
      />
      <input
        autoFocus={true}
        className="inputform  normal"
        id="inputKey"
        onChange={[Function]}
        placeholder="Enter your unique mnemonic signing key"
        value=""
      />
      <div
        className="error-message none"
      >
        
      </div>
    </form>
    <span
      className="submit-button-container"
    >
      <button
        className="button btn-primary "
        id="submit"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Submit
        </div>
      </button>
    </span>
  </div>,
]
`;

exports[`KeyModalContent renders description 1`] = `
Array [
  <h1>
    Test
  </h1>,
  <p>
    Test description
  </p>,
  <div
    className="key-input-container "
  >
    <form>
      <div
        className="label"
      />
      <input
        autoFocus={true}
        className="inputform  normal"
        id="inputKey"
        onChange={[Function]}
        placeholder="Enter your unique mnemonic signing key"
        value=""
      />
      <div
        className="error-message none"
      >
        
      </div>
    </form>
    <span
      className="submit-button-container"
    >
      <button
        className="button btn-primary "
        id="submit"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Submit
        </div>
      </button>
    </span>
  </div>,
]
`;
